src/cogitate.py

```diff
@@ -1,58 +1,272 @@
+    # Assess PyDriller access with link validator method
+    if repository == "INVALID" or link_validator(args["link"]) is False:
+        print("Cannot authenticate repository.")
+        progress_bar.finish()
+        return
+    else:
+        # Populate json file
+        try:
+            data_collection.collect_and_add_raw_data_to_json(
+                args["link"], "raw_data_storage"
+            )
+            progress_bar.next(1)
+            print("  Repository authenticated")
+            progress_bar.next(1)
+            print("  Raw Data Collected")
+        except BaseException:
+            print("Invalid repository link: " + args["link"])
+            progress_bar.finish()
+            return
+        # end of repository authentication
+        # calculate metrics to be used for team evaluation
+        issue_dict = {}
+        issue_dict = data_collection.retrieve_issue_data(
+            repository, args["state"], issue_dict
+        )
+        progress_bar.next(1)
+        print("  Issue Data Collected")
+        # calculate individual metrics
+        individual_metrics_dict = data_collection.calculate_individual_metrics()
+        progress_bar.next(1)
+        print("  Individual Data Calculated")
+        merged_dict = data_collection.merge_metric_and_issue_dicts(
+            individual_metrics_dict, issue_dict
+        )
+        # merge duplicate usernames if user requests to
+        if args["runmerge"]:
+            progress_bar.next(1)
+            print("  Merging Duplicate Usernames")
+            while True:
+                data_collection.print_individual_in_table(
+                    data_dict=merged_dict, headings=[],
+                )
+                name_to_keep = input("Please enter the username to keep:  ")
+                name_to_merge = input("Please enter the username to merge:  ")
+                merged_dict = data_collection.merge_duplicate_usernames(
+                    merged_dict, name_to_keep, name_to_merge
+                )
+                cont = input("Merge another username? (y/n)")
+                if cont.lower() == "y":
+                    pass
+                else:
+                    print("Ending Username merge...")
+                    break
 
-    # Temporary structure given issue retrieval is the only function
-    contributor_data = data_collection.initialize_contributor_data(
-        "contributor_data_template"
-    )
-    contributor_data = data_collection.retrieve_issue_data(
-        repository, args["state"], contributor_data
-    )
+        elif not args["runmerge"]:
+            progress_bar.next(1)
+            print(
+                "  Duplicate usernames unmerged"
+                + "\nMerging duplicate usernames is suggested, "
+                + "\nTo do so change '-rm' to 'y' in your command line arguments"
+            )
+        progress_bar.next(1)
+        print("  Merged Data Sets")
```

Comment:
Small change needed, this progress bar advancement should be in the merging section to avoid confusion.

Reply:
That progress bar is meant for the when the issue and individual dicts have been merged above, I just forgot to move them when I moved the function call earlier, sorry for the confusion!

It has been moved to the correct place now.